### ISSUE TITLE
Cannot use terraform-provider-postgresql version 1.13.0 (though 1.13.0-pre1 works fine)

### ISSUE THREAD

**Issue by izinovik** [cyrilgdn/terraform-provider-postgresql#93]
Hello.

I cannot use provider version 1.13.0. Terraform reports that it failed to query provider packages:
```
% head -n 13 main.tf
terraform {
  required_version = "~> 0.13"

  backend "gcs" {
    bucket = "my-terraform-state"
    prefix = "production/postgresql_dbs"
  }

  required_providers {
    postgresql = {
      source  = "cyrilgdn/postgresql"
      version = "~> 1.13.0"
    }
```

Here is what happens when I try to upgrade my provider plugin (previously used version 1.10.0):
```
% terraform init -upgrade
Upgrading modules...
...

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/vault...
- Finding cyrilgdn/postgresql versions matching "1.10.0, ~> 1.13.0"...
- Using previously-installed hashicorp/vault v2.20.0

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
cyrilgdn/postgresql: no available releases match the given constraints 1.10.0,
~> 1.13.0
```

I use Terraform version 0.14.11.
```
% terraform version -json
{
  "terraform_version": "0.14.11",
  "terraform_revision": "",
  "provider_selections": {
    "registry.terraform.io/cyrilgdn/postgresql": "1.13.0-pre1",
    "registry.terraform.io/hashicorp/vault": "2.20.0"
  },
  "terraform_outdated": true
}
```

### Expected Behavior
Provider plugin installs successfully.

### Actual Behavior
Provider plugin is not installed

### Steps to Reproduce
1. `terraform init -upgrade` with provider version constrain set to `1.13.0` or `~> 1.13.0`

**Comment by cyrilgdn** [comment]
Hi @izinovik ,

It's probably a configuration problem on your side, the following message:
```
- Finding cyrilgdn/postgresql versions matching "1.10.0, ~> 1.13.0"...
```
explains that you're looking for a version which is equal to 1.10.0 AND at least 1.13 which is not possible.

I'd say that you instantiate a module somewhere which have a `required_providers` requiring the 1.10.0 version.

For example, I can have the same message with the following code:
```hcl
$ cat main.tf
terraform {
  required_providers {
    postgresql = { 
      source  = "cyrilgdn/postgresql"
      version = "~> 1.13.0"
    }   
  }
}

module "test" {
  source = "./module"
}
```

```hcl
$ cat module/test.tf
terraform {
  required_version = "~> 0.13"

  required_providers {
    postgresql = {
      source  = "cyrilgdn/postgresql"
      version = "1.10.0"
    }
  }
}

resource "postgresql_database" "test" {
  name = "test"
}
```

```bash
$ terraform init
Initializing modules...
- test in module

Initializing the backend...

Initializing provider plugins...
- Finding cyrilgdn/postgresql versions matching "1.10.0, ~> 1.13.0"...
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider cyrilgdn/postgresql: no available releases match the given constraints 1.10.0, ~> 1.13.0
╵
```

**Comment by izinovik** [comment]
Thanks, @cyrilgdn. Definitely it was a problem on my side, I set postgresql provider version to 1.10.0 in my `postgres_db` module.
Sorry for the noise.
Closing the issue.
